source/app.d

```diff
@@ -261,21 +262,23 @@ void readSpaces() {
 }
 
 string readLabel() {
-	string label;
+	int firstColumn = column;
 	while (!eol()) {
 		char c = line[column++];
-		if (c >= '0' && c <= '9' || c == '_') {
-			label ~= c;
+		if (c >= '0' && c <= '9' || c == '_' || c == '?')
 			continue;
-		}
 		c &= 0xdf;
-		if (c >= 'A' && c <= 'Z') {
-			label ~= c;
+		if (c >= 'A' && c <= 'Z')
 			continue;
-		}
 		column--;
 		break;
 	}
+	string label = line[firstColumn .. column];
```

Comment:
The new lexing is likely faster. Unfortunately it makes the labels case-sensitive, which breaks compatibility.